tests/db/compact_db_migrations.py

```diff
@@ -0,0 +1,67 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+
+
+    Tool to compact db migration scripts
+"""
+import asyncio
+import importlib
+
+import asyncpg
+import pytest
+
+from inmanta.data.schema import CORE_SCHEMA_NAME, DBSchema
+
+PACKAGE_NAME = "versions_to_compact"
+
+
+async def test_compact_and_dump(postgres_db, database_name):
```

Comment:
This approach has the downside that you have to move the migration scripts you want to compact to a separate directory. As such, there is the risk of accidentally forgetting to one version. I would add a check here to verify that no such mistake was made, namely that the version present in the versions_to_compact package start at V1 up to a certain version vX. And that all versions left in the original versions directory are all strictly higher than vX.